source/common/upstream/upstream_impl.cc

```diff
@@ -1297,7 +1297,9 @@ ClusterInfoImpl::ClusterInfoImpl(
       codec_filter->mutable_typed_config()->PackFrom(
           envoy::extensions::filters::http::upstream_codec::v3::UpstreamCodec::default_instance());
     }
-    if (http_filters[http_filters.size() - 1].name() != "envoy.filters.http.upstream_codec") {
+    if (http_filters[http_filters.size() - 1].typed_config().type_url() !=
+        TypeUtil::descriptorFullNameToTypeUrl(
+            "envoy.extensions.filters.http.upstream_codec.v3.UpstreamCodec")) {
```

Comment:
Thanks for the contribution.

The type string could be a static string to avoid building it every time.

And how we handle the scenario where the `TypedStruct` is used? 

/wait-any

Reply:
Thanks, updated to handle TypedStruct and use a static string.